app/services/preview_asset_service.rb

```diff
@@ -32,7 +25,49 @@ def upload_asset(asset)
 
 private
 
-  def auth_bypass_id
-    PreviewAuthBypassService.new(edition).auth_bypass_id
+  def payload
+    auth_bypass_id = PreviewAuthBypassService.new(edition).auth_bypass_id
+    { draft: true, auth_bypass_ids: [auth_bypass_id] }
+  end
+
+  def update(asset)
+    GdsApi.asset_manager.update_asset(asset.asset_manager_id, payload)
+  end
+
+  def upload(asset)
+    upload = GdsApi.asset_manager.create_asset(
+      payload.merge(file: UploadedFile.new(asset)),
+    )
+
+    asset.update!(file_url: upload["file_url"], state: :draft)
+  end
+
+  # Used as a stand-in for a File / Rack::Multipart::UploadedFile object when
+  # passed to GdsApi::AssetManager#create_asset. The interface is required for
+  # uploading a file using the restclient we used in the backend.
+  #
+  # https://github.com/rest-client/rest-client/blob/master/lib/restclient/payload.rb#L181-L194
+  #
+  # This is done by delegating to 'io' which should implement the IO interface
+  # (e.g. StringIO) and adds methods to return filename, content_type and path.
+  class UploadedFile < SimpleDelegator
```

Comment:
I'm keen on having it close to the service, since it's the only place where it's used - time will tell if it's really generic ;-). Creating more top-level files feels more painful than keeping it where it is. One alternative might be to namespace it into a folder, since it's really a supporting class for the PreviewAssetService - we do this for GovspeakDocument.

How does that sound?